src/test/lib/test-async-iterable.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+// Set Symbol.asyncIterator on browsers without it
+if (typeof Symbol !== undefined && Symbol.asyncIterator === undefined) {
+  Object.defineProperty(Symbol, 'asyncIterator', {value: Symbol()});
+}
+
+export class TestAsyncIterable<T> implements AsyncIterable<T> {
+  /**
+   * A Promise that resolves with the next value to be returned by the
+   * async iterable returned from iterable()
+   */
+  private _nextValue: Promise<T> =
+      new Promise((res, _) => this._resolveNextValue = res);
```

Comment:
Inconsistent naming here. `res` is used vs `resolve` bellow. In general, I prefer full words over abbreviations.

Reply:
done